serverlessworkflow/modules/ROOT/pages/service-orchestration/orchestration-of-openapi-based-services.adoc

```diff
@@ -0,0 +1,423 @@
+      }
+   ]
+}
+----
+
+The `operation` attribute is a string composed by:
+
+1. The URI scheme that the engine uses to locate the specification file
+2. The location path of the resource
+3. The operation identifier (`operationId`), which you find in the link:{open_api_spec_url}#fixed-fields-7[OpenAPI specification file].
+
+Kogito supports the following URI schemes:
+
+* `classpath` for files located in the application project under the `src/main/resources` folder. The `classpath` is the default scheme when you don't define a scheme in the string. For example `classpath:/myopenapifile.yaml` or `/myopenapifile.yaml`. In this example, the file location is in `src/main/resources/myopenapifile.yaml` path
+* `file` for files located in the filesystem
+* `http` or `https` for remotely located files
+
+[IMPORTANT]
+====
+The OpenAPI specification files must be available only on build time. This is because, during runtime, Kogito leverages its internal code generation feature to make the REST calls. So {build-workflow-image-with-quarkus-cli}[once you build your application image], the Kogito engine won't access the files anymore.
+====
+
+== Getting Access to the OpenAPI Specification Files
+
+To get started, you must locate the REST services OpenAPI specification files.
+
+Usually, REST services expose their interface via the OpenAPI specification. For example, see the link:https://petstore3.swagger.io/api/v3/openapi.json[Petstore API].
+
+If the service you want to add to the workflow doesn't have the specification file, you can either create one or update the service to generate and expose it.
+
+The vast majority of the popular REST frameworks already support the OpenAPI specification generator. For example:
+
+* Java
+** link:{quarkus_swagger_url}[Quarkus Swagger]
+** link:{spring_swagger_url}[Spring Boot Docs]
+* Go
+** link:{go_swagger_url}[Go Swagger]
+* Python
+** link:{python_flasgger_url}[Flasgger]
+* .NET
+** link:{dotnet_swagger_url}[ASP.NET Core web API documentation with Swagger / OpenAPI]
+* TypeScript
+** link:{npm_swagger_url}[Swagger UI Express]
+* PHP
+** link:{php_swagger_url}[Swagger PHP]
+
+If you don't have access to the REST service to add the OpenAPI specification generator, you can use the link:{swagger_inspector_url}[Swagger Inspector] to generate the file. This tool will generate the specification file based on HTTP traffic.
+
+== Making REST Calls based on the OpenAPI Specification
+
+To make REST calls using OpenAPI specification files, you must define the function references and then use these functions in the workflow states.
+
+=== Defining the OpenAPI functions
+
+Once you have access to the OpenAPI specification files, you can create the function definitions in the workflow. You can use the link:{kogito_examples_base_url}/serverless-workflow-temperature-conversion[serverless-workflow-temperature-conversion] example application as a reference.
+
+.Procedure
```

Comment:
Not sure a title is working well here (like it does for images and tables etc.)? It seems to render as normal text.  Maybe ==== ?

Reply:
@ricardozanini we use (===) for section titles in Asciidoc. Since a procedure is part of the same title. For more information, see https://docs.asciidoctor.org/asciidoc/latest/sections/titles-and-levels/

For procedures, we follow the guidelines mentioned in the IBM style guide: https://www.ibm.com/docs/en/ibm-style?topic=format-procedures

Therefore, I'm following the same format in Serverless docs to be consistent. Please let me know if you need more details or have any questions.

Reply:
@krisv I think we can keep the `.Procedure` format, but maybe render it differently once we have a UI theme.

Reply:
Yes, I don't have anything against using the title, the concern is indeed related to how it's rendered right now, as it doesn't seem to stand out from normal text.  But it's a minor thing we can indeed address later.